ui/aura/window_tree_host_platform.h

```diff
@@ -56,7 +57,7 @@ class AURA_EXPORT WindowTreeHostPlatform : public WindowTreeHost,
 
   // Creates a ui::PlatformWindow appropriate for the current platform and
   // installs it at as the PlatformWindow for this WindowTreeHostPlatform.
-  void CreateAndSetDefaultPlatformWindow();
+  void CreateAndSetDefaultPlatformWindow(const client::WindowType type);
```

Comment:
What if we add a method called CreateAndSetPlatformWindowWithParameters(const Widget::InitParams& params), which will store Widget::InitParams, and then call CreateAndSetDefaultPlatformWindow. 

Then, PlatformWindowDelegate can declare getters we need such as GetWindowType, GetParentWindowWidget (this will be used instead of WaylandConnection::GetCurrentFocusedWindow) and etc?

Reply:
OK. I'll update.